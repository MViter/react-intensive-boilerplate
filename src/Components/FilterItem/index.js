// Core
import React, { Component } from 'react';

// Instruments
import { func, string } from 'prop-types';
import Styles from './styles.scss';

export default class FilterItem extends Component {


    static propTypes = {
        getNews:                   func.isRequired,
        getNewsFromDefinedSources: func.isRequired,
        sourceID:                  string.isRequired
    };

    constructor () {
        super();
        this.handleCheckboxClicking = ::this._handleCheckboxClicking;
        // this.getNewsFromDefinedSources = ::this._getNewsFromDefinedSources;
    }

    _handleCheckboxClicking (event) {
        //event.preventDefault();
        const checkBoxValue = event.target.value;
        const isChecked = event.target.checked;

        this.props.getNewsFromDefinedSources(checkBoxValue, this.props.getNews, isChecked);

    }

    render () {

        const { sourceID } = this.props;

        return (
            <section className = { Styles.container } >
                <div className = { Styles.filterItem } >
                    <input
                        type = 'checkbox'
                        //checked = {sourceID.indexOf(sourceIDs) !== -1}
                        value = { sourceID }
                        onClick = { this.handleCheckboxClicking }
                    />
                    <a href = '#'>{ sourceID }</a>
                </div>
            </section>
        );
    }
}
