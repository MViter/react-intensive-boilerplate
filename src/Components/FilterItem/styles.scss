.filterItem{
  a {
    text-decoration: none;
    color: #fff;
  }
}