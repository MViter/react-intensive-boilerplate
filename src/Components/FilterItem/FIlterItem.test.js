import React from 'react';
import FilterItem from './';
import dom from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
const getNewsMock = jest.fn();
const getNewsFromDefinedSourcesMock = jest.fn();

Enzyme.configure({ adapter: new Adapter() });

const renderTree = dom.create(
    <FilterItem getNews = { getNewsMock } getNewsFromDefinedSources = { getNewsFromDefinedSourcesMock } sourceID = 'abc-news' />
).toJSON();

const result = shallow(<FilterItem getNews = { getNewsMock } getNewsFromDefinedSources = { getNewsFromDefinedSourcesMock } sourceID = 'abc-news' />);

describe('<FilterItem/> component tests: ', () => {
    it('Component should be equal to its snapshot', () => {
        expect(renderTree).toMatchSnapshot();
    });

    test('Should have 1 input elenemt', () => {
        expect(result.find('input').length).toBe(1);
    });
});
