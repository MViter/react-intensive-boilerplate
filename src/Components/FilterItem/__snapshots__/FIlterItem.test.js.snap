// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterItem/> component tests:  Component should be equal to its snapshot 1`] = `
<section
  className={undefined}
>
  <div
    className={undefined}
  >
    <input
      onClick={[Function]}
      type="checkbox"
      value="abc-news"
    />
    <a
      href="#"
    >
      abc-news
    </a>
  </div>
</section>
`;
